lms/djangoapps/course_blocks/transformers/user_partitions.py

```diff
@@ -64,26 +64,25 @@ def collect(cls, block_structure):
             merged_group_access = _MergedGroupAccess(user_partitions, xblock, merged_parent_access_list)
             block_structure.set_transformer_block_field(block_key, cls, 'merged_group_access', merged_group_access)
 
-    def transform(self, usage_info, block_structure):
-        """
-        Mutates block_structure based on the given usage_info.
-        """
-        SplitTestTransformer().transform(usage_info, block_structure)
+    def transform_block_filter(self, usage_info, block_structure):
+        split_test_filter = SplitTestTransformer().transform_block_filter(usage_info, block_structure)
 
         user_partitions = block_structure.get_transformer_data(self, 'user_partitions')
-
         if not user_partitions:
-            return
+            return [block_structure.create_universal_filter()]
 
         user_groups = _get_user_partition_groups(
             usage_info.course_key, user_partitions, usage_info.user
         )
-        block_structure.remove_block_if(
+        group_access_filter = block_structure.create_removal_filter(
             lambda block_key: not block_structure.get_transformer_block_field(
                 block_key, self, 'merged_group_access'
             ).check_group_access(user_groups)
         )
 
+        split_test_filter.append(group_access_filter)
+        return split_test_filter
```

Comment:
Nit: rename `split_test_filter` to be something else since it ends up being a list of filters from *both* split_test and user_partitions.  Alternatively, move L68 down and append the split_test_filter at the end of the function.